crowbar_framework/app/controllers/backups_controller.rb

```diff
@@ -0,0 +1,170 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+class BackupsController < ApplicationController
+  skip_before_filter :enforce_installer
+  before_action :set_backup, only: [:destroy, :restore, :download]
+
+  #
+  # Backups
+  #
+  # Provides the restful api call for
+  # /utils/backup 	GET 	Returns a json list of available backups
+  def index
+    @backups = Backup.all
+
+    respond_to do |format|
+      format.html
+      format.json { render json: @backups }
+    end
+  end
+
+  #
+  # Backups
+  #
+  # Provides the restful api call for
+  # /utils/backup   POST   Trigger a backup
+  def create
+    @backup = Backup.new(backup_params)
+
+    respond_to do |format|
+      if @backup.save
+        format.json { head :ok }
+        format.html { redirect_to backups_path }
+      else
+        msg = I18n.t(".invalid_filename", scope: "backups.index")
+        format.json { render json: { error: msg }, status: :bad_request }
+        format.html do
+          flash[:alert] = msg
+          redirect_to backups_path
+        end
+      end
+    end
+  end
+
+  #
+  # Restore
+  #
+  # Provides the restful api call for
+  # /utils/backup/restore   POST   Trigger a restore
+  def restore
+    respond_to do |format|
+      if @backup.valid?
+        ret = @backup.restore
+        if ret[:status] == :ok
+          format.json { head :ok }
+          format.html { redirect_to backups_path }
+        else
+          format.html do
+            flash[:alert] = ret[:msg]
+            redirect_to backups_path
+          end
+          format.json { render json: { error: ret[:msg] }, status: :bad_request }
+        end
+      else
+        msg = I18n.t(".invalid_backup", scope: "backups.index")
+        format.json { render json: { error: msg }, status: :bad_request }
+        format.html do
+          flash[:alert] = msg
+          redirect_to backups_path
+        end
+      end
+    end
+  end
+
+  #
+  # Download
+  #
+  # Provides the restful api call for
+  # /utils/backup/download/:name/:created_at 	GET 	Download a backup
+  def download
+    respond_to do |format|
+      if @backup.exist?
+        format.any do
+          send_file(
+            @backup.path,
+            filename: @backup.filename
+          )
+        end
+      else
+        msg = I18n.t(".missing_backup", scope: "backups.index")
+        format.json { render json: { error: msg }, status: :bad_request }
+        format.html do
+          flash[:alert] = msg
+          redirect_to backups_path
+        end
+      end
+    end
+  end
+
+  #
+  # Upload
+  #
+  # Provides the restful api call for
+  # /utils/backup/upload   POST   Upload a backup
+  def upload
+    @backup = Backup.new(backup_upload_params)
+
+    respond_to do |format|
+      if @backup.save
+        format.json { head :ok }
+        format.html { redirect_to backups_path }
+      else
+        format.json {
```

Comment:
Style/BlockDelimiters: Avoid using `{...}` for multi-line blocks. (https://github.com/bbatsov/ruby-style-guide#single-line-blocks)